### PR TITLE
Added Groups API

### PR DESCRIPTION
**Pull request by ramakantar**
- Added API Definition
- Added Test Cases
- Added Examples

### CODE REVIEW

File: internal/mockserver/mockdata/groups.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
+package mockdata
+
+var (
+	// MockedListGroupsResponse mocked `ListGroups` endpoint response
+	MockedListGroupsResponse = []byte(`
+    {
+        "count": 12,
```
Comment (Colstuwjx) [inline internal/mockserver/mockdata/groups.go]: count = 12 ?

Comment (ramakantar) [reply internal/mockserver/mockdata/groups.go]: It's just a mocked data to test.

Comment (Colstuwjx) [reply internal/mockserver/mockdata/groups.go]: fine

**Comment by Colstuwjx** [comment]
Thanks.

**Pull request merged**